modules/core/src/v2/coins/eosutil/eosTextDecoder.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * @prettier
+ */
+import { StringDecoder } from 'string_decoder';
+
+/**
+ * Custom utf8 TextDecoder that uses StringDecoder. This is used to deserialize EOS transactions.
+ *
+ * OVC uses Electron to package their apps. Electron crashes with the global TextDecoder when decoding large inputs.
+ * Our work around is to implement our own TextDecoder using StringDecoder instead.
+ *
+ * Electron issue: https://github.com/electron/electron/issues/18733
+ */
+export class EosTextDecoder extends TextDecoder {
```

Comment:
why don't we use call it stringDecoder and allow it to be used in a generic way? Not sure why we need to tie this function to EOS.

Reply:
Updated - there are some limitations - this only supports utf8 and does not support streaming. but shouldn't be a problem.

added comment to make this clear.